app/src/main/java/in/ac/bits_pilani/goa/ard/activities/MainActivity.java

```diff
@@ -77,6 +125,41 @@ public void onClick(final View view) {
         toggle.syncState();
 
         navigationView.setNavigationItemSelectedListener(this);
+        navDrawerDBRef = FirebaseDatabase.getInstance().getReference(AHC.FDR_NAV_DRAWER);
+
+        final View headerView = navigationView.getHeaderView(0);
+        final ImageView navDrawerImage = ButterKnife.findById(headerView, R.id.nav_drawer_image);
+        final TextView navDrawerTitle = ButterKnife.findById(headerView, R.id.nav_drawer_title);
+        final TextView navDrawerSubtitle = ButterKnife.findById(headerView, R.id.nav_drawer_subtitle);
+
+        if (navDrawerTitleText != null) {
+            navDrawerTitle.setText(navDrawerTitleText);
```

Comment:
Yes my concern is only holding references to views. But I think it is fine for now..we can use Leakcanary later if we find app is lagging